code/modules/modular_computers/obj/item/computer.dm

```diff
@@ -0,0 +1,188 @@
+// This is the base type of computer
+// Other types expand it - tablets and laptops are subtypes
+// consoles use "procssor" item that is held inside it.
+/obj/item/modular_computer
+	name = "modular microcomputer"
+	desc = "A small portable microcomputer."
+	icon = 'icons/obj/computer.dmi'
+	icon_state = "laptop-open"
+	light_on = FALSE
+	integrity_failure = 0.5
+	max_integrity = 100
+	armor = list(MELEE = 0, BULLET = 20, LASER = 20, ENERGY = 100, BOMB = 0, BIO = 0, FIRE = 0, ACID = 0)
+	light_system = MOVABLE_LIGHT_DIRECTIONAL
+
+	/// Starting programs for this computer
+	var/list/datum/computer_file/program/starting_programs = list()
+	/// Our modular computer
+	var/datum/modular_computer_host/item/cpu = /datum/modular_computer_host/item
```

Comment:
I also don't really know how much I agree with this refactor in general, it seems to make it even more confusing because now you have to deal with certain things on the item (icon state, deconstructable, disk, cell) and other things on the computer host (like actually using said charge).

I should've said something sooner but I kept procrastinating on reading through this.